docs/guides/blocks/sync-tables/two-way.md

```diff
@@ -0,0 +1,293 @@
+
+<img src="../../../../images/two_way_enable.png" srcset="../../../../images/two_way_enable_2x.png 2x" class="screenshot" alt="Enabling edits on a sync table.">
+
+Select columns of the table will now be editable, and you can change cell values them as you would for a normal Coda table. Use the **Update rows** link to push the changes back to the data source.
+
+![type:video](../../../images/two_way.mp4){: .screenshot}
+
+
+## Adding two-way sync
+
+You can add support for two-way sync to any sync table, even after it's published. You'll have to adjust the schema used by your sync table and add a function that knows how to make the updates.
+
+
+### Annotate the schema
+
+Not every field that you sync from the external data source can be changed. To indicate which columns should be editable you need to annotate your schema by adding `mutable: true` to the corresponding properties.
+
+```{.ts hl_lines="7"}
+const TaskSchema = coda.makeObjectSchema({
+  properties: {
+    name: {
+      description: "The name of the task.",
+      type: coda.ValueType.String,
+      fromKey: "content",
+      mutable: true,
+    },
+    // ...
+  },
+  // ...
+});
+```
+
+Only top-level properties in the schema are eligible to be editable, since they are stored in distinct columns. Properties in nested objects, even if marked as mutable, will not be editable in the sync table. You can however still make a column containing an nested object editable, but you'll need to provide [autocomplete](#autocomplete) to generate a list of compatible objects.
+
+??? tip "Annotating a shared schema"
+
+    If the property's schema is reused in multiple columns, and only some of those are editable, you can use the [object spread operator][mdn_spread_object] to duplicate the schema and merge in the `mutable` field.
+
+    ```{.ts hl_lines="10-13"}
+    const UserSchema = coda.makeObjectSchema({
+      // ...
+    });
+
+    const TaskSchema = coda.makeObjectSchema({
+      properties: {
+        // The creator cannot be edited.
+        creator: UserSchema,
+        // The assignee can be edited.
+        assignee: {
+          ...UserSchema,
+          mutable: true,
+        },
+        // ...
+      },
+      // ...
+    });
+    ```
+
+
+### Write the update function
+
+Once the user has made their edits you'll need to figure out how to push them back to the external data source. This code is implemented in the `executeUpdate` function, which lives alongside the `execute` function in the sync formula.
+
+```{.ts hl_lines="9-11"}
+pack.addSyncTable({
+  name: "Tasks",
+  // ...
+  formula: {
+    // ...
+    execute: async function (args, context) {
+      // Pull down the data ...
+    },
+    executeUpdate: async function (args, updates, context) {
+      // Push up the changes ...
```

Comment:
Should this say `// Push up the changes and return the updated data ...`?

Reply:
Done.